content/2019.04.getting-started-with-elm.rst

```diff
@@ -0,0 +1,784 @@
+
+If your editor is well configured with ``elm-format``, you should see
+this as soon as you save it:
+
+.. code-block:: elm
+
+    module Main exposing (main)
+
+    import Html
+
+
+    main =
+        Html.text "Hello world!"
+
+If not, you can run ``elm-format`` manually on your file:
+
+.. code-block:: bash
+
+    ~/tutorial$ elm-format --yes Main.elm
+
+
+Opening your app in the browser
+-------------------------------
+
+One way to run your app is to use ``elm reactor``, the core app browser
+provided by the platform:
+
+.. code-block:: bash
+
+    ~/tutorial$ elm reactor
+    Go to <http://localhost:8000> to see your project dashboard.
+
+Then open http://localhost:8000/src/Main.elm in your favorite Web browser.
+
+
+Learning about the Elm virtual DOM
+==================================
+
+Virtual DOM functions to generate HTML are in the ``Html`` module.
+
+The ``Html`` module we used above to render some text also exposes many
+more functions for rendering HTML tags. You can import them all using:
+
+.. code-block:: elm
+
+    import Html exposing (..)
+
+Note that unlike with some other languages, the Elm compiler will
+complain if you try to import symbols already defined in the current
+module, which makes it actually useful and really enjoyable to use.
+
+You can then use ``text`` directly for instance:
+
+.. code-block:: elm
+
+    main = text "Hello world"
+
+The Virtual DOM HTML nodes are functions named after standard HTML
+tags, and take two parameters:
+
+- A list of attributes
+- A list of children
+
+If I want to create a ``div`` with a link it would look like this:
+
+.. code-block:: elm
+
+    module Main exposing (main)
+
+    import Html exposing (..)
+    import Html.Attributes exposing (..)
+
+
+    main =
+        div
+            [ class "container" ]
+            [ a
+                [ href "http://www.servicedenuages.fr/" ]
+                [ text "Blog" ]
+            ]
+
+We can also create a list of links in our div:
+
+.. code-block:: elm
+
+    module Main exposing (main)
+
+    import Html exposing (..)
+    import Html.Attributes exposing (..)
+
+
+    main =
+        div
+            [ class "container" ]
+            [ ul
+                [ class "links" ]
+                [ li
+                    []
+                    [ a
+                        [ href "http://www.servicedenuages.fr/" ]
+                        [ text "Blog" ]
+                    ]
+                , li
+                    []
+                    [ a
+                        [ href "http://www.elm-lang.org/" ]
+                        [ text "Elm lang" ]
+                    ]
+                ]
+            ]
+
+
+Adding some state
+=================
+
+Now that you know how to render your page in HTML, let's see how to
+write a program that handles events.
+
+The way Elm handles that is by having:
+
+- a ``Model``, an Elm record: a bit like a JavaScript object with
+  properties, that keep the state of the app
```

Comment:
I think this is two thoughts -- Model is an Elm record which is a bit like a JS object, and a Model is used to keep the state of the app. I think you could maybe restructure it a little bit to emphasize one or the other or maybe only focus on one?